src/server/cards/intrigueFirst/Tribute.js

```diff
@@ -0,0 +1,24 @@
+import Card from 'cards/Card';
+
+export default class Tribute extends Card {
+  static cost = { coin: 5 };
+  static types = new Set(['Action']);
+  async onPlay(player) {
+    const leftPlayer = player.game.playerOrder[
+      player.game.currentPlayerIndex + 1 === player.game.players.size ? 0 : player.game.currentPlayerIndex + 1
+    ];
+    const cards = leftPlayer.lookAtTopOfDeck(2);
+    player.game.log(`${leftPlayer.name} reveals ${cards.map(c => c.title).join(', ')}`);
+    if (cards[0]) {
```

Comment:
In order to keep repetition low, I would do
```js
const cardNames = new Set();
cards.forEach(card => cardNames.add(card.title));
for (let i = 0; i < cardNames.size; i++) {
  if (cards[0].types.has.....
```

Reply:
(sets cannot have duplicate elements)

Reply:
```js
const cardNames = [];
    for (let i = 0; i < cards.length; i++) {
      if (cardNames.includes(cards[i].title)) continue;
      cardNames.push(cards[i].title);
      if (cards[i].types.has('Action')) player.actions += 2;
...
```

I don't think a set is terribly useful here since there still has to be a check if there element exists in it already or not.

I mean you could have a set of objects like
```js
x = new Set();
cards.forEach(c => x.add({c.title, c.types});
x.forEach(c =>{
  if (c.types .has('Action')...
```

But that doesn't feel particularly better.